src/lib.rs

```diff
@@ -8,25 +7,57 @@ use std::ffi::{CStr, CString};
 pub mod bindings;
 pub use bindings::*;
 
-mod remap;
+pub mod cache;
+pub mod continuations;
+pub mod headers;
+mod helpers;
+pub mod remap;
+pub mod request;
+pub mod response;
+pub mod status;
+pub mod transaction;
+pub mod url;
 pub use remap::*;
 
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    Request(#[from] request::Error),
+
+    #[error(transparent)]
+    Response(#[from] response::Error),
+
+    #[error(transparent)]
+    Transaction(#[from] transaction::Error),
+
+    #[error(transparent)]
+    InvalidRawString(#[from] crate::helpers::RawStrError),
+}
+
 pub fn ts_debug(tag: &str, message: &str) {
     let t = CString::new(tag).unwrap_or_default();
     let s = CString::new(message).unwrap_or_default();
-    unsafe { TSDebug(t.as_ptr(), s.as_ptr()); }
+    unsafe {
+        TSDebug(t.as_ptr(), s.as_ptr());
+    }
 }
 
 pub fn ts_error(message: &str) {
     let s = CString::new(message).unwrap_or_default();
-    unsafe { TSError(s.as_ptr()); }
+    unsafe {
+        TSError(s.as_ptr());
+    }
 }
```

Comment:
Are these functions logging-related? Perhaps this could be documented by placing them in a separate module.